src/main.c

```diff
@@ -1555,6 +1578,21 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
         if (proxy_protocol_node != NULL &&
             (proxy_protocol = (int)h2o_configurator_get_one_of(cmd, *proxy_protocol_node, "OFF,ON")) == -1)
             return -1;
+
+        if (packet_timestamping_node != NULL) {
+            if ((*packet_timestamping_node)->type != YOML_TYPE_SCALAR) {
+                h2o_configurator_errprintf(cmd, node, "`packet-timestamping` must be a string");
+                return -1;
+            }
+            if (strcasecmp((*packet_timestamping_node)->data.scalar, "ON") == 0) {
+                packet_timestamping_enabled = 1;
+            } else if (strcasecmp((*packet_timestamping_node)->data.scalar, "OFF") == 0) {
+                packet_timestamping_enabled = 0;
+            } else {
+                h2o_configurator_errprintf(cmd, node, "value of `packet-timestamping` must be either of: ON,OFF");
+                return -1;
+            }
+        }
```

Comment:
```suggestion
        if (packet_timestamping_node != NULL &&
            (packet_timestamping = (int)h2o_configurator_get_one_of(cmd, *packet_timestamping_node, "OFF,ON")) == -1)
            return -1;
```

Proposed code is correct, but we do not need to check the type (as the argument of `h2o_configurator_parse_mapping` specifies the type as `:s`), and we can use `h2o_configurator_get_one_of` exactly the same way as we do for `proxy_protocol`.

Reply:
updated this, thanks!